### DISCUSSION TITLE
Should we move from CircleCI to GitHub Actions?

### DISCUSSION THREAD

**Discussion by ylemkimon** [KaTeX/KaTeX#2415] [Ideas]
GitHub Actions seem to work nice and pretty straightforward.

Pros:
* Nice integration with GitHub
* Single codebase/infrastructure
* Able to test on MacOS and Windows
* Easier to view results
* Annotation support
* Doesn't have problem with forked repo enabling CI

Cons:
* Not possible to reuse steps. The amount of code duplication is a bit high (https://github.com/actions/runner/issues/646)
* Harder to view artifacts, e.g., screenshot diffs and new screenshots. They are zipped and have to be downloaded
* Cannot skip if code is not changed, with required status checks enabled (but they only take <2 min. and GitHub Actions hour is unlimited)
* [ci skip] or [skip ci] doesn't work for pull requests

**Comment by edemaine** [comment]
Just to add: like CircleCI, public repositories are free for GitHub Actions.

This sounds like a win for simplicity of our CI infrastructure, so I think it's a good idea.

**Comment by ronkok** [comment]
I agree that GitHub Actions sounds like a win.
